docs/reference/components.cue

```diff
@@ -391,6 +395,30 @@ components: {
 		metrics?: #MetricOutput
 	}
 
+	#IAM: {
+		#Policy: {
+			_action:      !=""
```

Comment:
Why do we use `_` for some of these? I've seen that used for local variables in some config, but I would think that consumers of the docs would want / need the `action` and `service` here.

Reply:
@jszwedko they're variables that are using in the main definition.

https://github.com/timberio/vector/pull/5304/files#diff-54489cba2420df24a734d683666649ff3093d682b7e7d88fb777f5fecfc66754R406

Reply:
I do see that, I guess my question is, why shouldn't they also part of the data that would be exported?

Reply:
@jszwedko In this case, `action` *is* exported while `_action` is used as an input. I've done this for two reasons:

1. The full action name is a composite of the service name and the action name, e.g. the `logs:DescribeLogStream` value for `action`, where `logs` is the service and `DescribeLogStream` is the `_action`. Just a little does of DRY.
1. The full action name is constructed differently for the two platforms.

In terms of not exporting `service`, I just didn't see a reason to export that to the docs.

Reply:
Aaah, yes I see. I missed that `action` is exported below as the formatted `service:action`. I imagine that'll be sufficient for consumers of the docs. Thanks for the explanation!